src/taipy/gui/gui.py

```diff
@@ -437,6 +434,10 @@ def _update_var(
             self.__send_var_list_update(list(derived_modified), var_name)
 
     def __call_on_change(self, var_name: str, value: t.Any, on_change: t.Optional[str] = None):
+        suffix_var_name = ""
+        if "." in var_name:
+            suffix_var_name = var_name[var_name.index(".") + 1 :]
```

Comment:
use split ?
or at least cache the index result

Reply:
thank you!